### PR TITLE
Get connected applications for a local authenticator

### PR DESCRIPTION
**Pull request by kayathiri4**
### Proposed changes in this pull request
$subject

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java [hunk 1]
```diff
@@ -174,6 +175,21 @@ public abstract IdentityProvider[] getAllIdentityProviders(String tenantDomain)
     public abstract LocalAuthenticatorConfig[] getAllLocalAuthenticators(String tenantDomain)
             throws IdentityApplicationManagementException;
 
+    /**
+     * Get connected applications for a local authenticator.
+     *
+     * @param authenticatorId Authenticator ID.
+     * @param tenantDomain    Tenant domain.
+     * @param limit
```
Comment (kayathiri4) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java:183]: Update the comments

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 2]
```diff
@@ -95,15 +98,7 @@
...
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
```
Comment (kayathiri4) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:106]: Do not use `*`

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 3]
```diff
@@ -1344,6 +1366,40 @@ public ImportResponse importSPApplication(ServiceProvider serviceProvider, Strin
         return importResponse;
     }
 
+    private void validateResourceId(String resourceId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (StringUtils.isEmpty(resourceId)) {
+            String data = "Invalid argument: Authenticator resource ID value is empty";
```
Comment (DMHP) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:1373]: String data > String message

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 4]
```diff
@@ -4222,6 +4224,56 @@ public ConnectedAppsResult getConnectedApplications(String resourceId, int limit
...
+                    while (resultSet.next()) {
+                        connectedApps.add(resultSet.getString("UUID"));
+                    }
+                }
+            }
+            String sqlQuery = IdPManagementConstants.SQLQueries.LOCAL_AUTH_CONNECTED_APPS_TOTAL_COUNT_SQL;
+            try (PreparedStatement prepStmt = connection.prepareStatement(sqlQuery)) {
+                prepStmt.setString(1, authenticatorName);
+                prepStmt.setInt(2, tenantId);
+                prepStmt.setString(3, authenticatorName);
```
Comment (DMHP) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:4258]: Should not this be the identity provider name?

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 5]
```diff
@@ -4286,6 +4338,77 @@ private PreparedStatement createConnectedAppsSqlStatement(Connection connection,
...
+        String databaseProductName = connection.getMetaData().getDatabaseProductName();
+        if (databaseProductName.contains("MySQL")
+                || databaseProductName.contains("MariaDB")
+                || databaseProductName.contains("H2")) {
+            sqlQuery = IdPManagementConstants.SQLQueries.GET_CONNECTED_APPS_LOCAL_MYSQL;
+            prepStmt = connection.prepareStatement(sqlQuery);
+            prepStmt.setString(1, name);
+            prepStmt.setInt(2, tenantId);
+            prepStmt.setInt(3, tenantId);
+            prepStmt.setString(4, name);
```
Comment (DMHP) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:4356]: As per the query, it expects idp name. But here we have passed the authenticator name.

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java [hunk 6]
```diff
@@ -4286,6 +4338,77 @@ private PreparedStatement createConnectedAppsSqlStatement(Connection connection,
...
+            prepStmt.setInt(5, offset);
+            prepStmt.setInt(6, limit);
+        } else if (databaseProductName.contains("Oracle")) {
```
Comment (DMHP) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:4359]: Shall we have small private methods for each case?

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java (hunk 5 above)
Comment (kayathiri4) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java:4356]: We store those authenticator name in the DB as IDP name. So by changing the authenticator name to `identityProviderName1` with https://github.com/wso2/carbon-identity-framework/pull/4518#discussion_r1171149710 will work?

Comment (DMHP) [reply components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java]: Or else let's add a comment explaining it.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/IdentityProviderDAOImpl.java [hunk 7]
```diff
@@ -167,6 +168,22 @@ public List<LocalAuthenticatorConfig> getAllLocalAuthenticators()
...
     }
 
+    @Override
+    public ConnectedAppsResult getConnectedAppsOfLocalAuthenticator(String authenticatorId, int tenantId, Integer limit,
+                                                                    Integer offset)
+            throws IdentityApplicationManagementException {
+
+        IdentityProviderManager idpManager = IdentityProviderManager.getInstance();
+        try {
+            ConnectedAppsResult app = idpManager.getConnectedAppsForLocalAuthenticator(authenticatorId, tenantId,
```
Comment (DMHP) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/IdentityProviderDAOImpl.java:178]: Please add a meaningful variable name app

**Review by DMHP — DISMISSED** [review]

**Review by ChanikaRuchini — APPROVED** [review]

**Pull request merged**